src/IECoreHoudini/SOP_SceneCacheSource.cpp

```diff
@@ -311,8 +311,8 @@ void SOP_SceneCacheSource::loadObjects( const IECore::SceneInterface *scene, Ima
 	{
 		return;
 	}
-	
-	if ( scene->hasObject() && UT_String( scene->name() ).multiMatch( params.shapeFilter ) && tagged( scene, params.tagFilter ) )
+
+	if ( scene->hasObject() && UT_String( getFullScenePath( scene ) ).multiMatch( params.shapeFilter ) && tagged( scene, params.tagFilter ) )
```

Comment:
I think we'll get better performance by concatenating the names ourself as we recurse (e.g. the same thing we're doing with the transform), as it'll avoid a loop over the ancestors on each recursion.